### ISSUE TITLE
Simple fullscreen support

### ISSUE THREAD

**Issue by subhaze** [Microsoft/vscode#5344]
- VSCode Version: 1.0
- OS Version: OS X 10.10

I could be missing this in an option somewhere, but if not, it would be awesome if there was an option for "use_simple_full_screen" (like in Sublime Text) which just full screens VS Code in the current window and not create a new space like it does currently.

**Comment by egamma** [comment]
Did you try `View>Toggle Full Screen`

**Comment by subhaze** [comment]
@egamma yes, this works for full screening VS Code into a new space, but it's not the same a 'simple full screen' where VS Code would full screen without entering a new space on the desktop.

**Comment by dynamix** [comment]
that is one of the things I really miss compared to Sublime as well. 👍

**Comment by matmancini** [comment]
+1

**Comment by rogeriopvl** [comment]
+1

**Comment by rands0n** [comment]
+1

**Comment by richarddavenport** [comment]
All you have to do is hold down the option key when you click the full screen button and it changes to the old plus button. Is that what you are looking for?

**Comment by subhaze** [comment]
@richarddavenport nope, that does maximize the editor and is what I currently do. But, it doesn't get rid of the menu/window chrome and actually put the app into "full-screen" mode :(

**Comment by bpourriahi** [comment]
+1

**Comment by kibiz0r** [comment]
+1

**Comment by zachflower** [comment]
I thought I'd post an update to this thread, since it has had some eyes on it for a while now. Simple (pre-Lion) fullscreen support has just been added in electron/electron#10254 and will (probably) be released in Electron 1.8.x. While I'm sure it will be a little while before vscode fully tests and upgrades to that version of Electron (1.8.0 is still in beta, after all), some of you will be happy to hear that there is at least a light at the end of the tunnel ☺️

**Comment by calebporzio** [comment]
+1

**Comment by SHND** [comment]
Any news on this one guys?

**Comment by clankill3r** [comment]
Yeah if you have iTerm for example they have the option:

<img width="226" alt="screen shot 2018-01-01 at 23 44 10" src="https://user-images.githubusercontent.com/738650/34471587-dbb4b480-ef4d-11e7-8ac5-312b85ad54d8.png">

Which is really great cause it allows to have a window on top.
The thing I hate the most about the native fullscreen with vscode is that when I run my code it goes away to another window. Therefor I can not test my app and see the console at the same time.

**Comment by richarddavenport** [comment]
@clankill3r, if you are on a later version of macOS (not sure which one) you can long press the green fullscreen button and it will let you have two windows split in full screen mode. 

I know that still doesn't solve the overall goal of this issue, but it sounded like you do what I like to do and have my code on one side of the screen and the app running on another.

**Comment by Robula** [comment]
Going fullscreen means that if you move the cursor too far to the top of the screen, you Mac menu bar scrolls down and covers VSCode's tab bar. Really annoying when using the mouse to change tabs.

**Comment by o8e** [comment]
Any update on this? I'm very nearly considering switching back to Sublime for this. I don't want to. 🙍‍♂️

**Comment by zachflower** [comment]
@o8e This functionality is possible in Electron v1.8.*, but VSCode is currently on Electron v1.7.12. As soon as VSCode upgrades its Electron version, then a patch can be submitted that will provide pre-Lion fullscreen. Unfortunately, I'm not sure _when_ that is expected to happen.

**Comment by ignu** [comment]
👍 This is my hangup here.

Native fullscreen is extra clunky for people with multiple monitors or people who need to frequently switch apps. 

So, most developers.

**Comment by eighteyes** [comment]
iTerm has excellent non-native full screen support, but idk how accessible
that would be. Plus the global keyboard shortcut means super fast context
switching. I favor the iTerm model vs osx spaces.

**Comment by clankill3r** [comment]
@reedspool go to `System Preferences -> Accessibility -> Display -> [x] Reduce motion`.

**Comment by sineld** [comment]
+1

**Comment by bpasero** [comment]
I see no reason why we could not support this feature now that Electron 2.0.x landed.

**Comment by alexandcote** [comment]
@bpasero let me know if I can help in any way ✌️

**Comment by bpasero** [comment]
Blocked by https://github.com/electron/electron/issues/13135

**Comment by vivacenontroppo** [comment]
https://github.com/electron/electron/pull/14646

It's already fixed. Hope you guys will add this option soon.

**Comment by bpasero** [comment]
We are close, but there is https://github.com/Microsoft/vscode/pull/55267#issuecomment-427870573

**Comment by mesqueeb** [comment]
I just want to update from the original post, this is also relevant on `OS Version: macOS 10.14`

**Comment by bpasero** [comment]
This landed on master. To enable, configure `window.nativeFullscreen: false`

**Comment by mesqueeb** [comment]
@bpasero I get "Unknown configuration setting" error when I try to add that to my config file.

**Comment by bpasero** [comment]
@mesqueeb you will have to wait until tomorrow and then get our insiders build: https://code.visualstudio.com/insiders/

**Comment by ignu** [comment]
@a-eid it's not in the insider build yet. looks like the current commit is 4a29bd3731e40c07622bae and this is a few commits after that. hopefully tomorrow.

**Comment by bpasero** [comment]
We had to stop the insiders build due to a bug, so you will probably only see this next week Monday.

**Comment by ignu** [comment]
it's in insiders now!

@bpasero there's a 1 or 2 second pause when i switch to simpleFullScreen now that wasn't on my custom build. is that from https://github.com/Microsoft/vscode/pull/55267/commits/9667519b858db6076f800590f3d433fb51ba6149#diff-7c4f2973d00eb042eb19c3c3260ae078R523 ? also i don't quite understand what that was fixing.

**Comment by bpasero** [comment]
@ignu seems quite fast for me, I do not think my change has an impact. That was to fix an issue (that is still there actually) where Electron would return a 0px width for the document while entering or leaving fullscreen. But if you think your delay is caused by it, just comment it out and try it.

**Comment by bpasero** [comment]
Fyi the setting got renamed from `window.nativeFullscreen` to `window.nativeFullScreen`

**Comment by kibiz0r** [comment]
I am seeing some weird behavior when using multiple windows. Windows no longer rendering but still appearing in the Window menu, or windows completely closing without warning.

I'll see if I can get some solid repro steps, but if it's not something with my specific setup then I bet you can repro easily regardless. I had no extensions enabled, and my settings file isn't too crazy.

[settings.json.txt](https://github.com/Microsoft/vscode/files/2534880/settings.json.txt) (Github won't upload JSON files?)

**Comment by bpasero** [comment]
@kibiz0r if you can distill steps, file a separate issue please. if this feature is too broken, I will take it out again...

**Comment by ignu** [comment]
@kibiz0r @bpasero cmd+` is fine for me on multiple monitors. i even put instances on three monitors and even tried a combination of some in fullscreen and some windowed and can't reproduce. I'm on High Sierra (10.13.2)

**Comment by kibiz0r** [comment]
I'm just on one monitor. I suppose it could potentially be affected by multiple monitors. I'll do a repro today and try it with two monitors as well.

**Comment by kibiz0r** [comment]
1. Install VSCode Insiders (bdfe4fd41f6a5fe538197b3cba937230249a563a)
2. Save this as `~/Library/Application Support/Code - Insiders/User/settings.json`:
```
{
    "window.nativeFullScreen": false,
    "window.restoreFullscreen": true
}
```
3. Open VSCode Insiders
4. cmd+ctrl+f to fullscreen the window
5. cmd+n to create a new file, type some stuff in it
6. cmd+shift+n to open a new window
7. cmd+ctrl+f to fullscreen this new window
8. cmd+n to create a new file, type some other stuff in it
9. cmd+` to switch between windows; see that this works fine
10. Quit VSCode Insiders
11. Re-open VSCode Insiders
12. See that only one window from the last session is re-opened
13. cmd+shift+n to open a new window
14. cmd+ctrl+f to fullscreen this new window
15. cmd+n to create a new file, type some more stuff in it
16. cmd+` to switch between windows
17. See that the app alternates between switching windows immediately after pressing cmd+\`, then the next time requiring the user to press cmd+\` twice before it will switch
18. Open the Window menu; see that there is only one entry, "Welcome", despite having two windows open
19. cmd+shift+w to close the first window; see that the "Do you want to save?" dialog slides down from the top of the screen and the window closes
20. cmd+shift+w to close some invisible second window; see that the "Do you want to save?" dialog slides up from the bottom of the screen; see that this didn't close any visible window
21. cmd+shift+w to close a third window (the second visible window); see that the "Do you want to save?" dialog slides down from the top of the screen and the window closes

**Comment by bpasero** [comment]
@ignu I (and others: https://github.com/Microsoft/vscode/issues/63291) are seeing a strange behaviour where when entering simple fullscreen, the mouse acts up: it no longer changes its cursor when moving over UI elements, such as the explorer divider to resize it. I also see no hover showing up in the editor when hovering over code.

Given these issues (also https://github.com/Microsoft/vscode/issues/5344#issuecomment-436264187) I think I will have to disable this feature until the underlying issues (possibly in Electron?) are fixed.

**Comment by calebporzio** [comment]
Because it's not enabled by default and at least for me - it works well - can we keep the feature? This has been one of my most anticipated features yet.

**Comment by bpasero** [comment]
I found a way around it and the feature will remain, however we have to disable the custom titlebar on macOS if this feature is enabled due to issues with the mouse interaction.

**Comment by clankill3r** [comment]
For me it is bugged but I still love it.
The bugged fullscreen is still better then apple's horrible native fullscreen.

- when opening vscode the green plus sign does not work, only the shortcut does.
- when I use a second monitor and have my laptop closed and then unplug the monitor and open my laptop the window is in the wrong size (too large). Maybe other apps have this as well, I need to check that when I'm in the office.
- the red orange green buttons dissapear after exiting fullscreen, but who uses those anyway...

**Comment by vivacenontroppo** [comment]
But you all can just double click on title bar on VSCode and the result is the same. So why even bother to add bugged feature.

**Comment by clankill3r** [comment]
How is the result the same?

**Comment by ignu** [comment]
It's not the same. I mean, it might seem silly to obsess over the few pixels the menu bar takes up "maximized" in macOS, but I find the aesthetics awful and I'm in my editor too many hours a day.

Anyway @clankill3r I'm curious why that happens to you. I fixed that very issue and Simple Fullscreen VS Code should resize itself whenever the resolution changes. (I also seem to have the same flow as you when I take my MacBook home for the night.)

**Comment by clankill3r** [comment]
For me it's not the few pixels the menu bar takes up, for me the important
thing is that when I run my code a new window can come one top off VSCode.
With native fullscreen it goes to another desktop and opens the app there,
and then when I close that app it stays at that desktop. So I have to
switch manually to VSCode, where command + tab often does not work cause
that will go to whatever sceen I don't have in fullscreen. Neither does
command + ~ work cause that never goes to fullscreen applications.

Op wo 28 nov. 2018 om 00:44 schreef Len Smith <notifications@github.com>:
